src/components/DataTable/DataTable.js

```diff
@@ -120,6 +121,22 @@ export default class DataTable extends React.Component {
           key: this.state.sortHeaderKey,
         }),
       });
+      return;
+    }
+
+    const rowIds = this.state.rowIds.sort();
```

Comment:
I don't think this does exactly what you think it does. This will sort the IDs of the rows, not the data itself. It passes our tests because in the tests, we are using IDs of 'A' 'B' and 'C' and expecting that order, it's a coincidence. I think this passes tests for now but may lead to odd behaviour

Reply:
@ajdaniel just was using `sort` as a deterministic hash in this case so the equality check would work.

Reply:
Ah. I understand now!

Reply:
Like you commented in the other thread though, this simple heuristic probably doesn't capture all the use-cases 😞